### ISSUE TITLE
Reverse proxy example doesn't build on Windows

### ISSUE THREAD

**Issue by TransientResponse** [rejectedsoftware/vibe.d#1784]
Doing `dub init proxy_test -t vibe.d`, pasting in this source:

```
import vibe.appmain;
import vibe.http.proxy;
import vibe.http.server;


shared static this()
{
	auto settings = new HTTPServerSettings;
	settings.port = 8080;
	settings.bindAddresses = ["::1", "127.0.0.1"];

	listenHTTPReverseProxy(settings, "vibed.org", 80);
}
````

Results in a number of linker errors culminating in 

```
OPTLINK : Warning 134: No Start Address
Error: linker exited with status 527
dmd failed with exit code 527.
```

I've attached an unabridge dump of all the errors. There appear to be some (possibly Unicode-related) text garbling in there. 

Platform is Windows 10 1703 (Creator's Update) x64, DUB 1.2.1 and DMD 2.074.1. 

Full dump:
[errors.txt](https://github.com/rejectedsoftware/vibe.d/files/1082225/errors.txt)

**Comment by thaven** [comment]
Looks like dmd is trying to 'link' a single object file, and thus is missing relevant libs. (druntime/phobos/vibe-d)

Could this be related to `import vibe.appmain`, instead of using the VibeDefaultMain version?

**Comment by TransientResponse** [comment]
Welp, now I look dumb. Copying the `"versions": "VibeDefaultMain"` from the example's JSON to my project's JSON solved all my problems. No more undefined symbols, no more linker error. Builds and runs just fine. I guess I figured DUB would take care of things with the `-t vibe.d`. Is there a place I could look for documentation on what the version specifiers mean, particularly for vibe? I found [this](http://code.dlang.org/package-format?lang=json#version-specs) but I don't know that the version specifiers mean.
